BWTSolver.cs

```diff
@@ -0,0 +1,92 @@
+namespace Burrows_Wheeler_Transform;
+
+/// <summary>
+/// Class containing Burrows-Wheeler transform methods.
+/// </summary>
+public class BWTSolver
+{
+/// <summary>
+/// Rearranges a string with Burrows-Wheeler transform algorithm.
+/// </summary>
+/// <param name="input">A string to transform.</param>
+/// <returns>Transformed string.</returns>
+    public static (string, int) BurrowsWheelerTransform(string? input)
+    {
+        if (input == string.Empty || input == null)
+        {
+            Console.WriteLine("EMPTYSTRING");
+            return ("EMPTYSTRING", -1);
+        }
+
+        int[] indices = new int[input.Length];
+        for (int i = 0; i < input.Length; i++)
+        {
+            indices[i] = i;
+        }
+
+        BWTHelper.Sort(indices, input);
+        int end = Array.IndexOf(indices, 0);
+        char[] output = new char[input.Length];
```

Comment:
```suggestion
        var end = Array.IndexOf(indices, 0);
        var output = new char[input.Length];
```
Можно лишний раз не писать и не дублировать типы